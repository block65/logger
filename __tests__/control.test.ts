import { describe, expect, test } from '@jest/globals';
import { testVanillaLogger } from './helpers.js';

describe('Control', () => {
  test('Error Object', async () => {
    const [logger, logPromise] = testVanillaLogger();
    logger.error(Object.assign(new Error('hallo'), { debug: 'wooyeah' }));
    await expect(logPromise).resolves.toMatchSnapshot();
  });

  test('Error Object serialized on non-error level', async () => {
    const [logger, logPromise] = testVanillaLogger();
    logger.info(new Error('hallo'));
    await expect(logPromise).resolves.toMatchSnapshot();
  });
});
