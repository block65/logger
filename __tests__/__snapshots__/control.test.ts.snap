// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control Error Object 1`] = `
Array [
  Object {
    "data": Object {
      "err": Object {
        "debug": "wooyeah",
        "message": "hallo",
        "name": "Error",
        "stack": "Error: hallo
    at Object.<anonymous> (~/logger/__tests__/control.test.ts:22:32)",
      },
    },
    "err": [Error: hallo],
    "level": 50,
    "msg": "hallo",
    "time": 2009-02-13T23:31:30.000Z,
  },
]
`;

exports[`Control Error Object serialized on non-error level 1`] = `
Array [
  Object {
    "data": Object {
      "err": Object {
        "message": "hallo",
        "name": "Error",
        "stack": "Error: hallo
    at Object.<anonymous> (~/logger/__tests__/control.test.ts:30:17)
    at processTicksAndRejections (node:internal/process/task_queues:~:~)",
      },
    },
    "err": [Error: hallo],
    "level": 30,
    "msg": "hallo",
    "time": 2009-02-13T23:31:30.000Z,
  },
]
`;
